src/main/java/seedu/duke/command/EditTransactionCommand.java

```diff
@@ -0,0 +1,219 @@
+                isSuccess = handleGoalEdit(income, idx);
+            }
+        } else if (type.equals("out")) {
+            Expense expense = StateManager.getStateManager().getExpense(idx);
+            transactionDescription = StateManager.getStateManager().getExpense(idx)
+                    .getTransaction().getDescription();
+            if (getArg(DESCRIPTION_ARG) != null && !getArg(DESCRIPTION_ARG).isBlank()) {
+                String originalDescription = expense.getTransaction().getDescription();
+                StateManager.getStateManager().getExpense(idx)
+                        .getTransaction().setDescription(getArg(DESCRIPTION_ARG));
+                String newDescription = expense.getTransaction().getDescription();
+                isSuccess = !originalDescription.equals(newDescription);
+            } else if (getArg(AMOUNT_ARG) != null && !getArg(AMOUNT_ARG).isBlank()) {
+                Double originalAmount = expense.getTransaction().getAmount();
+                StateManager.getStateManager().getExpense(idx)
+                        .getTransaction().setAmount(Parser.parseNonNegativeDouble(getArg(AMOUNT_ARG)));
+                Double newAmount = expense.getTransaction().getAmount();
+                isSuccess = !originalAmount.equals(newAmount);
+            } else if (getArg(CATEGORY_ARG) != null && !getArg(CATEGORY_ARG).isBlank()) {
+                isSuccess = handleCategoryEdit(expense, idx);
+            }
+        }
+        if (!isSuccess) {
+            throw new DukeException(ERROR_MSG);
+        }
+        printSuccess(ui, transactionDescription, idx + 1); // idx + 1 for format to show to user
+    }
+
+    private int getTransactionMaxSize(String type) {
+        int maxSize = 0;
+        if (type.equals("in")) {
+            maxSize = StateManager.getStateManager().getIncomesSize();
+        } else if (type.equals("out")) {
+            maxSize = StateManager.getStateManager().getExpensesSize();
+        }
+        return maxSize;
+    }
+
+    private int parseIdx(int maxSize) throws DukeException {
+        int index = Integer.parseInt(getDescription());
+        if (index < 1 || index > maxSize) {
+            throw new DukeException(INVALID_IDX);
+        }
+        return index;
+    }
+    
+    private void printSuccess(Ui ui, String description, int idx) {
+        String type = getArg("type").toLowerCase();
+        String transactionType = type.equals("in") ? "income" : "expense";
+        String msg = "Successfully edited " + transactionType + " no." + idx + " " + description;
+        ui.print(msg);
+    }
+
+    private boolean handleGoalEdit(Income income, int idx) throws DukeException {
+        boolean isSuccess = false;
+        boolean foundNewGoal = false;
+        Goal originalGoal = income.getGoal();
+        String newGoalDescription = getArg(GOAL_ARG);
+        if (originalGoal.getDescription().equals(newGoalDescription)) {
```

Comment:
Goals and categories are not case-sensitive for other parts of the program. Maybe equalsIgnoreCase would be better?